### PR TITLE
New migration

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**

### CODE REVIEW

File: services/workflows/baseWorkflow.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+          new Error(`Workflow validation - missing fields in root workflow`)
+        );
+      }
+      workflow.activities.forEach((activity) => {
+        {
+          if (activity.hasOwnProperty("name")) {
+            if (typeof this[activity.name] !== "function") {
+              reject(
+                new Error(
+                  `Workflow validation - There is no activity for ${activity.name}`
```
Comment (asafMasa) [inline services/workflows/baseWorkflow.js:108]: add " for workflow x"

File: services/workflows/baseWorkflow.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  checkWorkflowValidation(workflow = this._workflow) {
+    return new Promise((resolve, reject) => {
+      if (
+        !this._helper.objectContainsFields(
+          workflow,
+          this._validator.workflowFields
+        )
+      ) {
+        reject(
+          new Error(`Workflow validation - missing fields in root workflow`)
```
Comment (asafMasa) [inline services/workflows/baseWorkflow.js:99]: add " for workflow x"

File: services/workflows/baseWorkflow.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
+"use strict";
+
+const jWorkflow = require("jWorkflow");
+const _ = require("lodash");
+
+module.exports.BaseWorkflow = class BaseWorkflow {
```
Comment (asafMasa) [inline services/workflows/baseWorkflow.js:6]: abstract

File: utils/writer.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+var ResponsePayload = function(code, payload) {
```
Comment (asafMasa) [inline utils/writer.js:1]: check if in use

File: workflows/default-workflow.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (asafMasa) [inline workflows/default-workflow.json:1]: rename to meaningful names or delete if not needed

File: api/swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,192 @@
+swagger: "2.0"
+info:
+  description: "api  for stratego service"
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply api/swagger.yaml]: fixed in d36b973

File: config/development.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+    "server": {
+        "port": 3009
+    },
+    "fileSystem": {
+        "workflowsPath": "./workflows"
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply config/development.json]: fixed in d36b973

File: services/handlers/dataHandler.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+module.exports = class DataHandler {
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply services/handlers/dataHandler.js:3]: javascript doesn't support "abstract".
it is the correct way to declare abstract class.

File: services/handlers/strategoHandler.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+      await this.configureAsync();
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  async configureAsync() {
+    let rootpath = path.resolve(config.fileSystem.workflowsPath);
+    try {
+      // await fs.statSync(rootpath);
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply services/handlers/strategoHandler.js]: fixed in d36b973

File: services/workflows/baseWorkflow.js (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply services/workflows/baseWorkflow.js:6]: javascript doesn't support "abstract".
it is the correct way to declare abstract class.

File: workflows/default-workflow.json (hunk 5 above)
Comment (CL-SHLOMIKONCHA) [reply workflows/default-workflow.json]: fixed in d36b973

File: services/handlers/strategoHandler.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+      const files = await fs.readdirAsync(rootpath);
+      console.log(files);
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply services/handlers/strategoHandler.js]: fixed in 5cf1dfd

File: utils/writer.js (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply utils/writer.js]: removed and fixed in 5cf1dfd

File: logger/logger.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
+const { MCLogger } = require("@map-colonies/mc-logger");
+const service = require('../package.json');
+
+const config = {
+  level: "info",
```
Comment (asafMasa) [inline logger/logger.js:5]: get config settings from config file (json)

File: package-lock.json [hunk 12]
```diff
@@ -1,9 +1,18 @@
 {
-  "name": "stratego",
+  "name": "workflow",
   "version": "1.0.0",
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
+    "@map-colonies/mc-logger": {
+      "version": "1.1.0",
+      "resolved": "https://artifactory.rnd-hub.com:443/artifactory/api/npm/Npm-IMFamily-virt/@map-colonies/mc-logger/-/mc-logger-1.1.0.tgz",
```
Comment (asafMasa) [inline package-lock.json:9]: delete package lock and remove rnd hub

File: services/WorkflowsService.js [hunk 13]
```diff
@@ -1,30 +1,44 @@
...
+const logger = container.get("logger");
+const workflowHandler = container.get("workflowHandler");
 /**
  * create workflow
  *
  * body Workflow create new workflow based on json schema
  * no response value expected for this operation
  **/
 exports.workflowsPOST = async function (args, res, next) {
+  const incWorkflow = args;
```
Comment (asafMasa) [inline services/WorkflowsService.js:18]: incoming

File: services/WorkflowsService.js [hunk 14]
```diff
@@ -50,18 +66,23 @@ exports.workflowsDELETE = async function (args, res, next) {
...
+    fileExists
+      ? await DataHandlerFileSystem.removeFile(
           workflowsPath,
           workflowName,
           "json"
         )
       : res.end("Not found");
 
-      //TODO: add logger
+    logger.info(
```
Comment (asafMasa) [inline services/WorkflowsService.js:77]: logger.info with request type and args before doing an action

File: services/handlers/dataHandlerFileSystem.js [hunk 15]
```diff
@@ -31,7 +32,7 @@ module.exports = class DataHandlerFileSystem extends DataHandler {
 
   writeFile(filePath, fileName, data, fileExtension) {
     fs.writeFile(`${filePath}/${fileName}.${fileExtension}`, data, (err) => {
-      if (err) console.log(err);
+      if (err) throw err;
```
Comment (asafMasa) [inline services/handlers/dataHandlerFileSystem.js:35]: {

File: services/workflows/baseWorkflow.js [hunk 16]
```diff
@@ -1,35 +1,42 @@
 "use strict";
 
+const workflowError = require("../../errors/workflowError");
+const container = require("../../containerConfig");
 const jWorkflow = require("jWorkflow");
-const _ = require("lodash");
+const config = require("config");
```
Comment (asafMasa) [inline services/workflows/baseWorkflow.js:6]: require("lodash"); -> check tree shaking

File: services/workflows/baseWorkflow.js [hunk 17]
```diff
@@ -0,0 +1,188 @@
...
+    return new Promise((resolve, reject) => {
+      if (
+        !this._helper.objectContainsFields(
+          workflow,
+          this._validator.workflowFields
+        )
+      ) {
+        reject(
+          new workflowError(
+            `Workflow validation - missing fields in root workflow`
```
Comment (asafMasa) [inline services/workflows/baseWorkflow.js:109]: add missing fields

File: logger/logger.js (hunk 11 above)
Comment (CL-SHLOMIKONCHA) [reply logger/logger.js]: fixed in  001241c

Comment (CL-SHLOMIKONCHA) [reply logger/logger.js]: fixed in  001241c

File: services/workflows/baseWorkflow.js (hunk 16 above)
Comment (CL-SHLOMIKONCHA) [reply services/workflows/baseWorkflow.js:6]: fixed in db7688b

File: services/handlers/dataHandlerFileSystem.js (hunk 15 above)
Comment (CL-SHLOMIKONCHA) [reply services/handlers/dataHandlerFileSystem.js]: fixed in 9560e9d

File: services/WorkflowsService.js (hunk 13 above)
Comment (CL-SHLOMIKONCHA) [reply services/WorkflowsService.js]: fixed in b2550e4

File: services/WorkflowsService.js (hunk 14 above)
Comment (CL-SHLOMIKONCHA) [reply services/WorkflowsService.js:83]: fixed in 0433386

File: package-lock.json (hunk 12 above)
Comment (CL-SHLOMIKONCHA) [reply package-lock.json]: fixed in 5dbd313

File: services/workflows/baseWorkflow.js (hunk 17 above)
Comment (CL-SHLOMIKONCHA) [reply services/workflows/baseWorkflow.js]: fixed in de00ef3

File: test/services/handlers/workflowHandler.spec.js [hunk 18]
```diff
@@ -0,0 +1,45 @@
+// const chai = require("chai");
```
Comment (asafMasa) [inline test/services/handlers/workflowHandler.spec.js:1]: why comment?

**Pull request merged**